### DISCUSSION TITLE
ptvo quirks counter to interface HA

### DISCUSSION THREAD

**Discussion by SergeyBoev** [zigpy/zha-device-handlers#2124] [Q&A]
Need count impulses from water meters. Made firmware for cc2530 via https://ptvo.info/ All work, but I can't integrate it to HA. Tried write  quirk for AnalogInput 0x000c. My head is demaged. Where I can find simple quirk for AnalogInput? Why HA don't recognize it by self. Or may be example with virtual clustes and attributes. I want to see link between HA objects and ZHA

{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.Router: 1>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress|RxOnWhenIdle|MainsPowered|FullFunctionDevice: 142>, manufacturer_code=4447, maximum_buffer_size=80, maximum_incoming_transfer_size=160, server_mask=0, maximum_outgoing_transfer_size=160, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=False, *is_full_function_device=True, *is_mains_powered=True, *is_receiver_on_when_idle=True, *is_router=True, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": 260,
      "device_type": "0xfffe",
      "in_clusters": [
        "0x0000",
        "0x000c"
      ],
      "out_clusters": [
        "0x0000"
      ]
    },
    "2": {
      "profile_id": 260,
      "device_type": "0xfffe",
      "in_clusters": [
        "0x000c"
      ],
      "out_clusters": []
    },
    "3": {
      "profile_id": 260,
      "device_type": "0xfffe",
      "in_clusters": [
        "0x0006",
        "0x0007"
      ],
      "out_clusters": [
        "0x0006"
      ]
    },
    "4": {
      "profile_id": 260,
      "device_type": "0xfffe",
      "in_clusters": [
        "0x0006",
        "0x0008",
        "0x000c"
      ],
      "out_clusters": [
        "0x0006"
      ]
    }
  },
  "manufacturer": "ptvo.info",
  "model": "bse.counter",
  "class": "zigpy.device.Device"
}

**Comment by Hedda** [comment]
@SergeyBoev  Start by opening a new issue with "Device support request" -> https://github.com/zigpy/zha-device-handlers/issues

There you provide the requested device signature, diagnostic information, logs, etc. which makes it easier for others to answer and help.

PS: This is in ZHA docs -> https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices
